pycoal/mineral.py

```diff
@@ -62,14 +62,15 @@ def __init__(self, library_file_name, class_names=None, threshold=0.0, in_memory
          "-spectral library '%s', -class names '%s', -threshold '%s', -in_memory '%s'" 
             %(library_file_name, class_names, threshold, in_memory))
 
-    def classify_image(self, image_file_name, classified_file_name):
+    def classify_image(self, image_file_name, classified_file_name, classifier_type=None):
         """
-        Classify minerals in an AVIRIS image using spectral angle mapper
+        Classify minerals in an AVIRIS image using specified classifier_type, if none defaults to spectral angle mapper
         classification and save the results to a file.
 
         Args:
             image_file_name (str):      filename of the image to be classified
             classified_file_name (str): filename of the classified image
+            classifier_type (str):      Classifier type used
```

Comment:
Augment this with the list of possible classifier options e.g. SAM, GML, KMeans, MDC (MahalanobisDistanceClassifier), etc. More classifiers can be seen at http://www.spectralpython.net/class_func_glossary.html#classification